### PR TITLE
Save events

### PR DESCRIPTION
**Pull request by zhoua0805**
Changes made:
- Add basic validation for add-event-form
- Create a select tag that deplays the names of the saved lists
- Allow users to save their current list to a new list with their choice of name
  - After the user clicks save, the current list will be emptied
  - Users can then access the saved list by selecting it in the select tag above
  - All new edits in the saved list will be automatically saved in the database

> 
After yesterday's meeting, we decided to make some small changes to the data structure: 

         users --> user1  --> name
                          --> email
                          --> phoneNumber

         events --> user1 --> currentList --> event 1
                                          --> event 2
                          --> savedList1  --> event 1
                                          --> event 2
                          --> savedList2 ...

         places --> user1 --> place1
                          --> place2

Next Steps:
- Start integrating with the places data 
- Allow users to rename their saved lists/ delete their saved lists (NOT REQUIRED FOR MVP)
- Don't allow users to delete their saved list by deleting every event (or give a warning when they attempt to delete the last event in the list) (NOT REQUIRED FOR MVP)

### CODE REVIEW

File: planet/src/main/webapp/js/itinerary.js [hunk 1]
```diff
@@ -49,19 +63,67 @@ function renderStartingLocation() {
     }
 }
 
+function renderListOptions() {
```
Comment (EkkiD) [inline planet/src/main/webapp/js/itinerary.js:65]: what does 'list' refer to? List of events? List of itineraries?

File: planet/src/main/webapp/js/itinerary.js [hunk 2]
```diff
@@ -49,19 +63,67 @@ function renderStartingLocation() {
...
 async function addEvent() {
     const eventName = document.getElementById('add-event-name').value;
     const eventAddress = document.getElementById('add-event-address').value;
     const eventDuration = document.getElementById('add-event-duration').value;
     const userId = firebase.auth().currentUser.uid; 
+    const listName = document.getElementById('list-options').value;
+
+    // Validate the input fields
+    if ((!eventDuration) || (!eventAddress) || (!eventDuration)) {
+        alert('Please make sure to fill out all fields correctly');
```
Comment (EkkiD) [inline planet/src/main/webapp/js/itinerary.js]: We should give the user more specific information about what they messed up. Specifically which fields and what is wrong with that field. If you want, you can split this into a validateEventInput() function to keep this code path clean.

File: planet/src/main/webapp/js/itinerary.js [hunk 3]
```diff
@@ -15,26 +15,40 @@
...
+window.openAddEventForm = openAddEventForm;
+window.closeAddEventForm = closeAddEventForm;
+window.openSaveEventsForm = openSaveEventsForm;
+window.closeSaveEventsForm = closeSaveEventsForm;
 window.handleStartingLocationChange = handleStartingLocationChange;
 window.renderStartingLocation = renderStartingLocation;
+window.handleListOptionChange = handleListOptionChange;
 window.addEvent = addEvent;
+window.saveEvents = saveEvents;
 window.generateItinerary = generateItinerary;
```
Comment (emilyzhaoo) [inline planet/src/main/webapp/js/itinerary.js]: LGTM with nit: remove extra line

**Review by emilyzhaoo — APPROVED** [review]

File: planet/src/main/webapp/js/itinerary.js [hunk 4]
```diff
@@ -49,19 +63,67 @@ function renderStartingLocation() {
...
+}
+
+function handleListOptionChange() {
+    const listName = document.getElementById ('list-options').value;
+    sessionStorage.setItem('listName', listName);
+    // You can only save currentList as other lists
+    if (listName !== 'currentList') {
+        document.getElementById('save-events-button').style.display = 'none';
+    }else{
+        document.getElementById('save-events-button').style.display = 'inline-block';
```
Comment (EkkiD) [inline planet/src/main/webapp/js/itinerary.js:97]: How do you save a  list as itself? E.g. I have a list called "test", and I want to save changes to it.

**Review by EkkiD — CHANGES_REQUESTED** [review]

File: planet/src/main/webapp/js/itinerary.js (hunk 4 above)
Comment (zhoua0805) [reply planet/src/main/webapp/js/itinerary.js:97]: Changes will be automatically saved if users are on other lists (just like how currentList is automatically updated). The save button is for exporting currentList. Hope that makes sense!

File: planet/src/main/webapp/js/itinerary.js (hunk 1 above)
Comment (zhoua0805) [reply planet/src/main/webapp/js/itinerary.js:65]: It's the list of events. We're not saving itineraries in the back end

File: planet/src/main/webapp/js/itinerary.js (hunk 2 above)
Comment (zhoua0805) [reply planet/src/main/webapp/js/itinerary.js]: Okay!

**Review by EkkiD — APPROVED** [review]
LGTM

File: planet/src/main/webapp/js/itinerary.js (hunk 3 above)
Comment (zhoua0805) [reply planet/src/main/webapp/js/itinerary.js]: Fixed!

**Pull request merged**